lib/src/metta/environment.rs

```diff
@@ -0,0 +1,193 @@
+    ///   [Environment] will be configured using the OS-Specific platform configuration files.
+    ///
+    /// Depending on the host OS, the config directory locations will be:
+    /// * Linux: ~/.config/metta/
+    /// * Windows: ~\AppData\Roaming\TrueAGI\metta\config\
+    /// * Mac: ~/Library/Application Support/io.TrueAGI.metta/
+    ///
+    /// TODO: Repeat this documentation somewhere more prominent, like the top-level README
+    pub fn new() -> Self {
+        Self {
+            env: Environment::new(),
+            no_cfg_dir: false,
+        }
+    }
+
+    /// Sets (or unsets) the working_dir for the environment
+    pub fn set_working_dir(mut self, working_dir: Option<&Path>) -> Self {
+        self.env.working_dir = working_dir.map(|dir| dir.into());
+        self
+    }
+
+    /// Sets the `config_dir` that the environment will load.  A directory at the specified path will
+    /// be created its contents populated with default values, if one does not already exist
+    pub fn set_config_dir(mut self, config_dir: &Path) -> Self {
+        self.env.config_dir = Some(config_dir.into());
+        if self.no_cfg_dir {
+            panic!("Fatal Error: set_config_dir is incompatible with set_no_config_dir");
+        }
+        self
+    }
+
+    /// Configures the Environment not to load nor create any config files
+    pub fn set_no_config_dir(mut self) -> Self {
+        self.no_cfg_dir = true;
+        if self.env.config_dir.is_some() {
+            panic!("Fatal Error: set_config_dir is incompatible with set_no_config_dir");
+        }
+        self
+    }
+
+    /// Adds additional include paths to search for MeTTa modules
+    ///
+    /// NOTE: The most recently added paths will have the highest search priority, save for the `working_dir`,
+    ///   and paths returned first by the iterator will have higher priority within the same call to add_include_paths.
+    pub fn add_include_paths<P: Borrow<Path>, I: IntoIterator<Item=P>>(mut self, paths: I) -> Self {
+        let mut additional_paths: Vec<PathBuf> = paths.into_iter().map(|path| path.borrow().into()).collect();
+        additional_paths.extend(self.env.extra_include_paths);
+        self.env.extra_include_paths = additional_paths;
+        self
+    }
+
+    /// Initializes the shared platform Environment, accessible with [platform_env]
+    ///
+    /// NOTE: This method will panic if the platform Environment has already been initialized
+    pub fn init_platform_env(self) {
+        PLATFORM_ENV.set(self.build()).expect("Fatal Error: Platform Environment already initialized");
+        common::init_logger(false);
+    }
+
+    /// Returns a newly created Environment from the builder configuration
+    ///
+    /// NOTE: Creating owned Environments is usually not necessary.  It is usually sufficient to use the [platform_env] method.
+    pub fn build(self) -> Environment {
+
+        let mut env = self.env;
+
+        if !self.no_cfg_dir {
```

Comment:
The reason I ended up with this design was that the logic is actually tri-state, and I wanted the API to be very explicit.

The possible states are:
1. User will accept the OS-supplied default config directory
2. User wants to specify a custom config directory
3. User wants to tell MeTTa not to use any config directory whatsoever, and to not touch the file system at all.

I felt that third option was qualitatively different from the other two, and deserved a separate flag.

Another option would be an Enum with an embedded Path, which is what I would have gone for in pure Rust, but that has a lot of surface area because there is no similarly-behaved language construct in C nor in Python.